File: src/main/java/frc/robot/commands/drivetrain/ArcadeDrive.java
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

package frc.robot.commands.drivetrain;

import org.team997coders.spartanlib.math.MathUtils;

import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.OI;
import frc.robot.subsystems.DriveTrain;

public class ArcadeDrive extends CommandBase {

  double left, right;
  public ArcadeDrive() {
    addRequirements(DriveTrain.getInstance());
  }

  @Override
  public void initialize() {
    // DriveTrain.getInstance().playOrchestra("Megalovania.chrp");
    DriveTrain.getInstance().stopOrchestra();
  }

  @Override
  public void execute() {
    double forward = -OI.getInstance().gamepad1.getRawAxis(1);
    double turn = OI.getInstance().gamepad1.getRawAxis(4);

    forward = MathUtils.deadband(forward, 0.1);
    turn = MathUtils.deadband(turn, 0.1);

    forward *= 0.7;
    if (turn < 0) turn = -Math.pow(turn, 2);
    else turn = Math.pow(turn, 2);

    left = forward + turn;
    right = forward - turn;
    
    // DriveTrain.getInstance().setMotors(left, right);
    DriveTrain.getInstance().simpleAccelControl(left, right);
  }

  @Override
  public void end(boolean interrupted) {
    DriveTrain.getInstance().setMotors(0, 0);
  }

  @Override
  public boolean isFinished() {
    return false;
  }
}

File: src/main/kotlin/frc/robot/commands/drivetrain/ArcadeDrive.kt
package frc.robot.commands.drivetrain

import com.ctre.phoenix.motorcontrol.NeutralMode
import edu.wpi.first.wpilibj.XboxController
import edu.wpi.first.wpilibj2.command.CommandBase
import frc.robot.Constants
import frc.robot.OI
import frc.robot.subsystems.Drivetrain

class ArcadeDrive: CommandBase() {

  init {
    addRequirements(Drivetrain)
  }

  override fun initialize() {
    Drivetrain.setNeutralMode(NeutralMode.Brake)
  }

  override fun execute() {

    System.out.println("Uhhh: " + OI.mGamepad1.getRawAxis(1))

    val f = Constants.deadband(-OI.mGamepad1.getRawAxis(1) * 0.7, 0.05)
    val t = Constants.deadband(OI.mGamepad1.getRawAxis(4) * 0.4, 0.05)

    val left = f + t
    val right = f - t

    Drivetrain.setMotors(left, right)
  }

  override fun isFinished(): Boolean {
    return false;
  }

  override fun end(interrupted: Boolean) {
    Drivetrain.setMotors(0.0, 0.0)
  }

}